ieee754r/fmaxmagd32.c

```diff
@@ -0,0 +1,61 @@
+/* The fmaxmag functions determine the value of their argument 
+   whose magnitude is the maximum of  the  magnitudes  of  the  arguments:
+   the value of x if |x| > |y|, y if |x| < |y|, and fmax(x,y) otherwise.
+
+   Copyright (C) 2007-2019 Free Software Foundation, Inc.
+
+   This file is part of the Decimal Floating Point C Library.
+
+   Author(s): Pedro Caldeira <caldeira@linux.ibm.com>
+
+   The Decimal Floating Point C Library is free software; you can
+   redistribute it and/or modify it under the terms of the GNU Lesser
+   General Public License version 2.1.
+
+   The Decimal Floating Point C Library is distributed in the hope that
+   it will be useful, but WITHOUT ANY WARRANTY; without even the implied
+   warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See
+   the GNU Lesser General Public License version 2.1 for more details.
+
+   You should have received a copy of the GNU Lesser General Public
+   License version 2.1 along with the Decimal Floating Point C Library;
+   if not, write to the Free Software Foundation, Inc., 59 Temple Place,
+   Suite 330, Boston, MA 02111-1307 USA.
+
+   Please see libdfp/COPYING.txt for more information.  */
+
+#ifndef _DECIMAL_SIZE
+#  define _DECIMAL_SIZE 32
+#  include <decimal32.h>
+#endif
+
+#include <decContext.h>
+#include <decNumber.h>
+#include <math.h>
+#include <ieee754r_private.h>
+
+#define FUNCTION_NAME fmaxmag
+
+#include <dfpmacro.h>
+
+DEC_TYPE
+INTERNAL_FUNCTION_NAME (DEC_TYPE x, DEC_TYPE y)
+{
+  decContext context;
+  decNumber dn_result;
+  DEC_TYPE result;
+  decNumber dn_x;
+  decNumber dn_y;
+
+  FUNC_CONVERT_TO_DN (&x, &dn_x);
+  FUNC_CONVERT_TO_DN (&y, &dn_y);
+
+  decContextDefault (&context, DEFAULT_CONTEXT);
+  decNumberMaxMag(&dn_result, &dn_x, &dn_y, &context);
+
+  FUNC_CONVERT_FROM_DN (&dn_result, &result, &context);
+
+  return result;
```

Comment:
I think this should also implemented without explicitly falling back to decnumber.

This implementation can be moved into sysdeps/soft-dfp/